### DISCUSSION TITLE
Please start including real changelog in github releases

### DISCUSSION THREAD

**Discussion by ssbarnea** [python-jsonschema/jsonschema#984] [Ideas]
If do not ask for too much, it would be really useful to get a real changelog include in releases. Currently they are not really too informative. I will include two screenshots from automatic email I get from releases on jsonschema and from another pypi package, so you would better understand the difference:

![](https://sbarnea.com/ss/Screen-Shot-2022-08-17-15-53-15.48.png)
![](https://sbarnea.com/ss/Screen-Shot-2022-08-17-15-53-36.07.png)

Also compare github release pages from https://github.com/ansible/ansible-lint/releases and https://github.com/python-jsonschema/jsonschema/releases 

The changelog is generated and updated automatically on ansible-lint project due to the use of https://github.com/release-drafter/release-drafter - which is very easy to setup and tune. I would be happy to assist if needed as I have being using it for 3+ years on more than 20 libraries.

**Comment by Julian** [comment]
Right now I'm using the [automatic GitHub release note generator](https://github.com/python-jsonschema/jsonschema/blob/c2e86eea66fcd0991524fcd06f76e1b233b3faff/.github/workflows/ci.yml#L190) -- which was working fine/nicely for a few releases and then weirdly stopped working for reasons I don't understand and didn't look into.

But I'm happy to consider alternatives, albeit I've tried lots of the automatic changelog tools and found issues with each :/

**Reply by ssbarnea** [reply]
AFAIK, Github one requires you to manually press the "generate changelog" button to refresh the data. It is very similar to release drafter, so similar that I suspect that it was used as "do something like this!". Still, github one is far less flexible, requiring manual intervention.

Maybe we can have a talk in the following days, i can show you how it works.

Yes, no such tool is perfect. The final experience also depends on what kind of requirements one might decide to be critical and on which one you are willing to being to the current limitations. On most of my project, I already have the drafter version ready and doing a release is only "edit draft, check the open discussion thread and press publish". The SemVersion is already there, generated based on the nature of the included changes, I just LOVE it!
